src/views/common/DashOverview.tsx

```diff
@@ -51,7 +51,12 @@ export const DashOverviewIntro: React.FC<{
     MIN_SAFE_HEALTH_FACTOR
   );
 
-  const limitAmount = balance && maxAmount?.lt(balance) ? maxAmount : balance;
+  const limitAmount =
+    balance && maxAmount?.lt(balance)
+      ? maxAmount.gt(0)
+        ? maxAmount
+        : constants.Zero
+      : balance;
```

Comment:
I just realized, leaving `balance` here is a problem- it should default to 0 if its inputs are undefined, otherwise you can go negative if the max is invalid and the balance itself is a `BigNumber` that is otherwise invalid.
```suggestion
      : undefined;
```

Reply:
Discussed with Luigy- this is intentional; comments or self-explanatory code needed in the future. Approving as bugfix for now.